src/core/qgsvectorlayer.h

```diff
@@ -1376,6 +1376,21 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     int addTopologicalPoints( const QgsPointXY &p );
```

Comment:
Can you deprecate this one?

Reply:
Done, and also for `QgsMapToolEdit::addTopologicalPoints( const QVector<QgsPointXY> &vertices )`
Thank you.

Reply:
@nyalldawson I reverted my commit since it's too big for this PR. I prefered create a dedicated PR https://github.com/qgis/QGIS/pull/31171